### ISSUE TITLE
If AtomicRateLimiter#acquirePermission() can't get permission within timeout, how about return false immediately?

### ISSUE THREAD

**Issue by injae-kim** [resilience4j/resilience4j#2194]
https://github.com/resilience4j/resilience4j/blob/2ceed9563c661ef30bb0ef893d6c539acc5e6b48/resilience4j-ratelimiter/src/main/java/io/github/resilience4j/ratelimiter/internal/AtomicRateLimiter.java#L319-L323

On `AtomicRateLimiter`, I found that it blocks `waitForPermission(timeoutInNanos)` although user can't acquire permission within `timeout`.

```java
        if (canAcquireInTime) {
            return waitForPermission(nanosToWait);
        }
        // waitForPermission(timeoutInNanos); // removed, immediately return false;
        return false;
```
How about not blocking `waitForPermission(timeoutInNanos)` and **return false immediately**?

If this code is intended, please just close this issue 😅 thank you!

**Comment by RobWin** [comment]
hi thank you.
Means you would like to introduce a new configuration parameter `waitUntilTimeout` which controls this behavior?

**Comment by injae-kim** [comment]
> new configuration parameter `waitUntilTimeout` which controls this behavior?

@RobWin yes! I want to add new `waitUntilTimeout(default: true)` config for user

**Comment by storozhukBM** [comment]
You can use **reservePermission** method for this purpose
